### PR TITLE
[SAIC-827] Skip creating security groups for undefined tenants

### PR DESCRIPTION
**Pull request by SVilgelm**
Skip a security group if its tenant is not found.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:196: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/identity/keystone.py:580: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'type_record'
cloudferrylib/os/identity/keystone.py:581: [W0612(unused-variable), KeystoneIdentity._get_roles_sql_request] Unused variable 'inher_tmp'
cloudferrylib/os/identity/keystone.py:627: [W0141(bad-builtin), KeystoneIdentity._get_user_tenants_roles_by_db] Used builtin function 'map'. Using a list comprehension can be clearer.
cloudferrylib/os/identity/keystone.py:627: [W0110(deprecated-lambda), KeystoneIdentity._get_user_tenants_roles_by_db] map/filter on lambda could be replaced by comprehension
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:814: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:196: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:814: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:196: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:810: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/1962/

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:196: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:810: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.identity.keystone
cloudferrylib/os/identity/keystone.py:194: [W0221(arguments-differ), KeystoneIdentity.deploy] Arguments number differs from overridden 'deploy' method
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:810: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/network/neutron.py [hunk 1]
```diff
@@ -546,20 +546,16 @@ def convert_security_groups(sec_gr, cloud):
...
-            rule_info = NeutronNetwork.convert(rule, cloud, 'rule')
-            security_group_rules.append(rule_info)
+        tenants = identity_res.get_tenants_list_dict()
 
         result = {
             'name': sec_gr['name'],
             'id': sec_gr['id'],
             'tenant_id': sec_gr['tenant_id'],
-            'tenant_name': get_tenant_name(sec_gr['tenant_id']),
+            'tenant_name': tenants[sec_gr['tenant_id']],
```
Comment (aguzikova) [inline cloudferrylib/os/network/neutron.py]: Looks like you re-wrote get_tenants_func(), but you don't use it anymore.
     def get_tenants_func(self):
         tenants = self.get_tenants_list_dict()

Anyway... 
tenants[sec_gr['tenant_id']]
seems to fail if there is no such tenant ID...
I would suggest to use something like this:
tenants.get(sec_gr['tenant_id'], 'admin')
 
         def func(tenant_id):
             return tenants.get(tenant_id, 'admin')
 
         return func

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module tests.cloudferrylib.os.network.test_neutron
tests/cloudferrylib/os/network/test_neutron.py:48: [E1002(super-on-old-class), NeutronTestCase.setUp] Use of super on an old style class
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:813: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: tests/cloudferrylib/os/network/test_neutron.py [hunk 2]
```diff
@@ -604,6 +604,9 @@ def test_get_security_groups(self):
 
         self.neutron_mock_client().list_security_groups.return_value = \
             fake_secgroups_list
+        self.identity_mock.get_tenants_list_dict.return_value = {
```
Comment (aguzikova) [inline tests/cloudferrylib/os/network/test_neutron.py]: looks not needed anymore)

Comment (SVilgelm) [inline tests/cloudferrylib/os/network/test_neutron.py]: Done

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.network.neutron
cloudferrylib/os/network/neutron.py:179: [W0221(arguments-differ), NeutronNetwork.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/network/neutron.py:284: [W0221(arguments-differ), NeutronNetwork.create_port] Arguments number differs from overridden 'create_port' method
cloudferrylib/os/network/neutron.py:813: [W0221(arguments-differ), NeutronNetwork.get_security_groups] Arguments number differs from overridden 'get_security_groups' method

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by aguzikova** [comment]
+1 looks good

**Comment by abochkarev** [comment]
+1

**Comment by roman-verchikov** [comment]
Can be merged

**Pull request merged**